### PR TITLE
refactor(docs-browser): routerless docs-browser

### PR DESCRIPTION
**Pull request by izzappel**
- react router v6 does not allow nested routing
- support routerless docs-browser to be able to upgrade

Refs: TOCDEV-5060

### CODE REVIEW

File: packages/docs-browser/src/modules/routing/sagas.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import pathToRegexp from 'path-to-regexp'
```
Comment (izzappel) [inline packages/docs-browser/src/modules/routing/sagas.js:1]: `path-to-regexp` deps should be added to root package.json

File: packages/docs-browser/src/modules/routing/sagas.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import {takeLatest, all, put} from 'redux-saga/effects'
+
+import * as actions from './actions'
+
+const pathRegex = '/docs/:model/:key/:view(list|detail)'
+
+export function* setParams({payload: {path}}) {
```
Comment (izzappel) [inline packages/docs-browser/src/modules/routing/sagas.js:8]: maybe this logic could be shared with `getPathInfo` in admin

File: packages/docs-browser/src/modules/routing/sagas.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  const pathParts = []
+  const re = pathToRegexp(pathRegex, pathParts)
+  const res = re.exec(path)
+
+  let params = {}
+  if (res !== null) {
+    params = pathParts.reduce((acc, pathPart, idx) => {
+      return {
```
Comment (rzueger) [inline packages/docs-browser/src/modules/routing/sagas.js]: (`return` not necessary as only statement)

File: packages/docs-browser/src/utils/withRouter.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+const mapStateToProps = state => ({
+  path: state.docs.routing.path,
+  params: state.docs.routing.params
+})
+
+export const withInternalRouter = connect(mapStateToProps, mapActionCreators)
+
+export const withReactRouter = WrappedComponent => {
+  return props => {
```
Comment (rzueger) [inline packages/docs-browser/src/utils/withRouter.js:19]: (`return` not necessary as only statement)

File: packages/docs-browser/src/components/DocsBrowser/Content/RoutesContent.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import {Route, Switch} from 'react-router-dom'
+
+import DocsView from '../../DocsView'
+import DocumentView from '../../DocumentView'
+
+const RoutesContent = ({navigationStrategy, handleSearchChange, emitAction, openFileDialog, searchMode, docsKey}) => {
+  return (
```
Comment (rzueger) [inline packages/docs-browser/src/components/DocsBrowser/Content/RoutesContent.js:9]: (`return` not necessary as only statement)

File: packages/docs-browser/README.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ React-registry name: `docs-browser`
 
 | Name                   | Mandatory | Description
 |------------------------|:---------:|-------------
-| `history`              |           | The history to use (if not provided, a hash history is created)
+| `routerType`           |           | routerless (navigation is not reflected in browser history e.g. Move Action) | inherit (take over routing context from parent e.g. Admin) | standalone (creates new routing context e.g. Widget)
```
Comment (rzueger) [inline packages/docs-browser/README.md:13]: (maybe mark "standalone" as default here?)

**Review by rzueger — APPROVED** [review]

**Pull request merged**